posthog/models/organization.py

```diff
@@ -191,18 +193,27 @@ def validate(self, *, user: Any = None, email: Optional[str] = None) -> None:
         _email = email or (hasattr(user, "email") and user.email)
 
         if _email and _email != self.target_email:
-            raise ValueError("This invite is intended for another email address.")
+            raise exceptions.ValidationError(
+                f"This invite is intended for another email address ({mask_email_address(self.target_email)}).",
```

Comment:
This message should include `_email`, in case a user can log in to an external account using the correct email, but can't sign up to PostHog with that email, because it's only an alias and the primary email is different (experienced this with GitHub - invited `dev@twixes.com`, could log into GitHub with it, but could not sign up into PostHog as my primary GH email is `michal@matloka.com`).